tests/integration/common_workflows/test_filesystem_workflows.py

```diff
@@ -0,0 +1,216 @@
+        batching_regex=re.compile(BATCHING_REGEX),
+    )
+
+
+@pytest.fixture
+def filesystem_monthly_batch_definition_descending(file_system_asset: CSVAsset) -> BatchDefinition:
+    return file_system_asset.add_batch_definition(
+        name="monthly",
+        partitioner=PartitionerYearAndMonth(
+            column_name="CHANGE ME",
+            sort_ascending=False,
+        ),  # The column name will be going away with the file-based fluent batch definition API
+        batching_regex=re.compile(BATCHING_REGEX),
+    )
+
+
+def _create_test_cases():
+    """Create our test cases.
+    With each flow, we want to see that we can validate an entire asset,
+    as well as subsets of the asset, including sorting and using batch parameters.
+    """
+    return [
+        pytest.param(
+            gxe.ExpectColumnDistinctValuesToEqualSet(
+                column=COLUMN_NAME, value_set=VALUES_FOR_MOST_RECENT_DATE
+            ),
+            "filesystem_whole_table_batch_definition",
+            None,
+            id="no batching regex - takes the last file",
+        ),
+        pytest.param(
+            gxe.ExpectColumnDistinctValuesToEqualSet(
+                column=COLUMN_NAME, value_set=VALUES_FOR_MOST_RECENT_DATE
+            ),
+            "filesystem_monthly_batch_definition",
+            None,
+            id="ascending",
+        ),
+        pytest.param(
+            gxe.ExpectColumnDistinctValuesToEqualSet(
+                column=COLUMN_NAME, value_set=VALUES_FOR_OLDEST_DATE
+            ),
+            "filesystem_monthly_batch_definition_descending",
+            None,
+            id="descending",
+        ),
+        pytest.param(
+            gxe.ExpectColumnDistinctValuesToEqualSet(
+                column=COLUMN_NAME, value_set=VALUES_ON_MY_FAVORITE_MONTH
+            ),
+            "filesystem_monthly_batch_definition",
+            MY_FAVORITE_DAY,
+            id="batch params",
+            marks=[pytest.mark.xfail(reason="Fix in V1-299", strict=True)],
+        ),
+    ]
+
+
+@pytest.mark.parametrize(
+    ("expectation", "batch_definition_fixture_name", "batch_parameters"),
+    _create_test_cases(),
+)
+@pytest.mark.filesystem
+def test_batch_validate_expectation(
+    expectation: gxe.Expectation,
+    batch_definition_fixture_name: str,
+    batch_parameters: Optional[Dict],
+    request: pytest.FixtureRequest,
+) -> None:
+    """Ensure Batch::validate(Epectation) works"""
+    batch_definition = request.getfixturevalue(batch_definition_fixture_name)
+    batch = batch_definition.get_batch(batch_parameters=batch_parameters)
+
+    result = batch.validate(expectation)
+    row_count_result = batch.validate(gxe.ExpectTableRowCountToEqual(value=10))
+
+    assert row_count_result.success
+    assert result.success
+
+
+@pytest.mark.parametrize(
+    ("expectation", "batch_definition_fixture_name", "batch_parameters"),
+    _create_test_cases(),
+)
+@pytest.mark.postgresql
```

Comment:
I think this mark and below should be changed to `pytest.mark.filesystem`.

Reply:
Good call, and thanks for the reminder on the top-level pytestmarks!